applications/API/models.py

```
@@ -0,0 +1,625 @@
+
+@receiver(post_save, sender=InsultReview)
+def increment_report_count(sender, instance, created, **kwargs):
+    """
+    Increments the report count for an insult when a new InsultReview is created.
+
+    This signal handler updates the reports_count field of the related Insult instance
+    whenever a new InsultReview is saved. It ensures the report count reflects the
+    current number of reviews associated with the insult.
+    """
+    if created:
+        if instance.insult_id:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)